trieste/acquisition/function/function.py

```diff
@@ -87,7 +87,7 @@ def update_acquisition_function(
         tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
         mean, _ = model.predict(dataset.query_points)
         eta = tf.reduce_min(mean, axis=0)[0]
-        function = probability_below_threshold(model, eta)
+        function.update(eta)  # type: ignore
```

Comment:
(can you also add a `tf.debugging.Assert(isinstance(function, probability_below_threshold), [tf.constant([])])`?)